source/_posts/2015-12-10-sometimes-its-good-to-be-shy.markdown

```diff
@@ -0,0 +1,242 @@
+---
+layout: post
+title: "Sometimes, it's good to be shy"
+category:
+comments: true
+author:
+  name: Rodrigo Ponce de León
+  email: rodrigo@wyeworks.com
+  twitter_handle: rponce_89
+  github_handle: rodrigopdl
+  image: /images/team/rodrigo-pdl.jpg
+  description: Ruby on Rails developer at WyeWorks. Beatlemaniac. Guitar freak.
+---
+
+> Life doesn't stand still.
+
+> Neither can the code that we write. In order to keep up with today's
+> near-frantic pace of change, we need to make every effort to write code
+> that's as loose--as flexible--as possible. Otherwise we may find our
+> code quickly becoming outdated, or too brittle to fix, and may
+> ultimately be left behind in the mad dash toward the future.
+
+This quote, taken from the highly regarded *"[The Pragmatic Programmer](https://pragprog.com/book/tpp/the-pragmatic-programmer)"* book,
+depicts beautifully an ongoing, everyday challenge software developers
+face: writing flexible enough code which smoothlessly adapts to change.
+
+What [Andy Hunt](https://twitter.com/PragmaticAndy) and [Dave Thomas](https://twitter.com/pragdave) are telling us is that
+software we develop, just as much as the world we live in, resides in a
+constant state of evolution. Despite our best efforts, requirements *will*
+change, and we'll be better off if our solution is prepared for it.
+
+<!--more-->
+
+In our quest to write flexible code, three concepts arise which I think
+help tremendously in achieving our ultimate goal, or at least they clear
+the path towards the consummation of this never ending struggle.
+
+## DRY.. DRY... DRY...
+
+> [Every piece of knowledge must have a single, unambiguous,
+> authoritative representation within a system.](https://pragprog.com/book/tpp/the-pragmatic-programmer)
+
+The acronym *DRY* stands for *"Don't Repeat Yourself"*, that is,
+not doing what I did on this section's title. When crafting software
+solutions, this translates to *ACD*: *"Avoid Code Duplication!"*.
+
+Code repetition aggravates the maintainability nightmare software
+developers are generally used to. Incoming changes affecting a duplicated piece
+of knowledge in our application would need to be taken care of in more than one
+place of our codebase. On complex systems, making these changes effectively
+constitutes a daunting task, which may lead to unexpected and hard to find
+bugs.
+
+This is why keeping your code *DRY* can help tremendously in creating flexible
+systems. When a shift in requirements occurs, or a bug is reported in
+production, you'll know exactly *where* to make the change.
```

Comment:
I don't think devs will always know *exactly where* make a change in those cases. If the code was done following the DRY principle, changing in *only one part of the code* would be enough to solve the problem.

In order words, I agree with the fact that only one part of the code should be changed in those cases, but I wouldn't state that I will always know where is the part of the code (it will depend on the size of the system, my skills, time working on the project and so on).